### DISCUSSION TITLE
tok2vec -> embed  extra fields not permitted

### DISCUSSION THREAD

**Discussion by jump155** [explosion/spaCy#9473] [Help: Coding & Implementations] [answered]
I am trying to use predictions from preceding components according to the [example](https://spacy.io/usage/training#config-components-listeners)
spacy Version: 3.1.3

I want ner component to use attributes from parser component.

Here is my config
```
[paths]
train = null
dev = null
vectors = null
init_tok2vec = null

[system]
gpu_allocator = null
seed = 0

[nlp]
lang = "ru"
pipeline = ["tok2vec","morphologizer","parser","ner"]
batch_size = 100
disabled = []
before_creation = null
after_creation = null
after_pipeline_creation = null
tokenizer = {"@tokenizers":"spacy.Tokenizer.v1"}

[components]

[components.morphologizer]
source = "output/model-rus"
replace_listeners = ["model.tok2vec"]

[components.morphologizer.model]
@architectures = "spacy.Tagger.v1"
nO = null

[components.morphologizer.model.tok2vec]
@architectures = "spacy.Tok2VecListener.v1"
width = ${components.tok2vec.model.encode.width}

[components.ner]
factory = "ner"
incorrect_spans_key = null
moves = null
update_with_oracle_cut_size = 100

[components.ner.model]
@architectures = "spacy.TransitionBasedParser.v2"
state_type = "ner"
extra_state_tokens = false
hidden_width = 64
maxout_pieces = 2
use_upper = true
nO = null

[components.ner.model.tok2vec]
@architectures = "spacy.Tok2VecListener.v1"
width = ${components.tok2vec.model.encode.width}
upstream = "*"

[components.ner.model.tok2vec.embed]
@architectures = "spacy.MultiHashEmbed.v1"
width = 256
attrs = ["NORM","PREFIX","SUFFIX","SHAPE", "DEP"]
rows = [5000,2500,2500,2500,2500]
include_static_vectors = true

[components.parser]
factory = "parser"
learn_tokens = false
min_action_freq = 30
moves = null
update_with_oracle_cut_size = 100

[components.parser.model]
@architectures = "spacy.TransitionBasedParser.v2"
state_type = "parser"
extra_state_tokens = false
hidden_width = 128
maxout_pieces = 3
use_upper = true
nO = null

[components.parser.model.tok2vec]
@architectures = "spacy.Tok2VecListener.v1"
width = ${components.tok2vec.model.encode.width}
upstream = "*"

[components.tok2vec]
factory = "tok2vec"

[components.tok2vec.model]
@architectures = "spacy.Tok2Vec.v2"

[components.tok2vec.model.embed]
@architectures = "spacy.MultiHashEmbed.v2"
width = ${components.tok2vec.model.encode.width}
attrs = ["ORTH","SHAPE"]
rows = [5000,2500]
include_static_vectors = true

[components.tok2vec.model.encode]
@architectures = "spacy.MaxoutWindowEncoder.v2"
width = 256
depth = 8
window_size = 1
maxout_pieces = 3

[corpora]
[corpora.dev]
@readers = "spacy.Corpus.v1"
path = ${paths.dev}
max_length = 0
gold_preproc = false
limit = 0
augmenter = null

[corpora.train]
@readers = "spacy.Corpus.v1"
path = ${paths.train}
max_length = 0
gold_preproc = false
limit = 0
augmenter = null

[training]
dev_corpus = "corpora.dev"
train_corpus = "corpora.train"
seed = ${system.seed}
gpu_allocator = ${system.gpu_allocator}
dropout = 0.1
accumulate_gradient = 1
patience = 1600
max_epochs = 0
max_steps = 20000
eval_frequency = 200
frozen_components = ["morphologizer"]
annotating_components = ["parser"]
before_to_disk = null

[training.batcher]
@batchers = "spacy.batch_by_words.v1"
discard_oversize = false
tolerance = 0.2
get_length = null

[training.batcher.size]
@schedules = "compounding.v1"
start = 100
stop = 1000
compound = 1.001
t = 0.0

[training.logger]
@loggers = "spacy.ConsoleLogger.v1"
progress_bar = false

[training.optimizer]
@optimizers = "Adam.v1"
beta1 = 0.9
beta2 = 0.999
L2_is_weight_decay = true
L2 = 0.01
grad_clip = 1.0
use_averages = false
eps = 0.00000001
learn_rate = 0.001

[training.score_weights]
pos_acc = 0.17
morph_acc = 0.17
morph_per_feat = null
dep_uas = 0.17
dep_las = 0.17
dep_las_per_type = null
sents_p = null
sents_r = null
sents_f = 0.0
ents_f = 0.33
ents_p = 0.0
ents_r = 0.0
ents_per_type = null

[pretraining]

[initialize]
vectors = ${paths.vectors}
init_tok2vec = ${paths.init_tok2vec}
vocab_data = null
lookups = null
before_init = null
after_init = null

[initialize.components]

[initialize.tokenizer]
```

But when I start training, I get the Error: 
✘ Config validation error
ner.model.tok2vec -> embed   extra fields not permitted

I tried removing come fields and blocks, but nothing helped. 
Could you, please, say, what have I made wrong?

**Comment by ljvmiranda921** [comment, marked as answer]
Hi @jump155 , you can run 

```sh
spacy debug config path/to/config.cfg
```

to better get a sense of what's missing / wrong in your config. Then you can start debugging from there 👍

**Reply by jump155** [reply]
Hi, @ljvmiranda921 ! Thank you!

There is the same response. 
It says that I should remove 'embed' field from 'component.ner.model.tok2vec', but documentation says, that I must specify attributes from previous component in component.*.model.tok2vec.embed
Here it is:

`~$ spacy debug config config.cfg

============================= Config validation =============================
✘ Config validation error
ner.model.tok2vec -> embed   extra fields not permitted

{'@architectures': 'spacy.Tok2VecListener.v1', 'width': 256, 'upstream': '*', 'embed': {'@architectures': 'spacy.MultiHashEmbed.v1', 'width': 256, 'attrs': ['NORM', 'PREFIX', 'SUFFIX', 'SHAPE', 'DEP'], 'rows': [5000, 2500, 2500, 2500, 2500], 'include_static_vectors': True}}`


Here is example from documentation:
`
[components.tagger.model.tok2vec.embed]
@architectures = "spacy.MultiHashEmbed.v1"
width = ${components.tagger.model.tok2vec.encode.width}
attrs = ["NORM","DEP"]rows = [5000,2500]
include_static_vectors = false

[training]
annotating_components = ["parser"]`

**Reply by ljvmiranda921** [reply]
Hi @jump155 , not familiar with your use-case, but from the config alone, you might need to use a [`spacy.Tok2Vec.v2`](https://spacy.io/api/architectures#Tok2Vec) instead of a [`spacy.Tok2VecListener.v1`](https://spacy.io/api/architectures#Tok2VecListener) in the `@architectures` parameter of your `[components.ner.model.tok2vec]`. The former has an `encode` and `embed` parameter, while the latter (the listener), only needs `width` and `upstream` (hence why you're getting the "extra fields not permitted" message). I suggest [checking this spaCy project](https://github.com/explosion/projects/tree/v3/pipelines/tagger_parser_predicted_annotations) for more info

**Reply by jump155** [reply]
@ljvmiranda921 Thank you! It worked for me.
